### PR TITLE
Updated to support IPv6

### PR DESCRIPTION
**Pull request by Kage1**
Added IPv6 support for each command, also updated the readme with additional details on usage.

### CODE REVIEW

**Comment by Kage1** [comment]
One additional thing to note, I had to change the format of the port command with a space instead of : since ipv6 uses : as separators.

File: tinystatus [hunk 1]
```diff
@@ -24,14 +24,24 @@ check(){
...
-    if [ ${ctype} = 'http' ]; then
-        statuscode="$(curl -sSkLo /dev/null -H "${useragent}" -m "${timeout}" -w "%{http_code}" "${host}" 2> "${tmp}/ko/${name}.error")"
-    elif [ ${ctype} = 'port' ]; then
-        error="$(nc -w "${timeout}" -zv $(echo "${host}" | sed 's,:, ,') 2>&1)"
+    if [ ${ctype} = 'http4' ]; then
+        statuscode="$(curl -4sSkLo /dev/null -H "${useragent}" -m "${timeout}" -w "%{http_code}" "${host}" 2> "${tmp}/ko/${name}.error")"
+    elif [ ${ctype} = 'http6' ]; then
+        statuscode="$(curl -6sSkLo /dev/null -H "${useragent}" -m "${timeout}" -w "%{http_code}" "${host}" 2> "${tmp}/ko/${name}.error")"
+    elif [ ${ctype} = 'port4' ]; then
+        error="$(nc -4w "${timeout}" -zv $(echo "${host}" | sed 's, , ,') 2>&1)"
```
Comment (bderenzo) [inline tinystatus:32]: Only netcat-openbsd support ipv4 and ipv6. For instance alpine linux nc does not support this option. At the moment I don't know how to solve this problem.

**Comment by bderenzo** [comment]
Thanks for your PR, I'll look this evening to see how to fix the nc problem.

**Comment by Kage1** [comment]
A couple of solutions I can think of.

1. Instead of replacing the existing commands, add the 4 & 6 versions like in the PR. For those who want the specific ones can use them, assuming there OS supports it.
2. Adding a note to the docs about the supported nc versions and leave the changes as they are.

**Comment by afro-coder** [comment]
`curl -v telnet://target ip address:desired port number`
Could we probably use this instead of nc or is it not a good thing because of telnet being bad.

**Comment by bderenzo** [comment]
> `curl -v telnet://target ip address:desired port number`
> Could we probably use this instead of nc or is it not a good thing because of telnet being bad.

I have already thought of using`curl`, but it does not support zero-I/O mode and need a timeout.
I have implemented ipv6 support based on Kage1 work, I will merge it this evening.

File: tinystatus (hunk 1 above)
Comment (alexpovel) [reply tinystatus:32]: [Alpine Linux does have the OpenBSD variant](https://pkgs.alpinelinux.org/package/edge/main/x86_64/netcat-openbsd), but it doesn't seem to support ipv4/ipv6, as you said:

```sh
$ nc --help # Alpine 3.12
BusyBox v1.31.1 () multi-call binary.

Usage: nc [OPTIONS] HOST PORT  - connect
nc [OPTIONS] -l -p PORT [HOST] [PORT]  - listen

        -e PROG Run PROG after connect (must be last)
        -l      Listen mode, for inbound connects
        -lk     With -e, provides persistent server
        -p PORT Local port
        -s ADDR Local address
        -w SEC  Timeout for connects and final net reads
        -i SEC  Delay interval for lines sent
        -n      Don't do DNS resolution
        -u      UDP mode
        -v      Verbose
        -o FILE Hex dump traffic
        -z      Zero-I/O mode (scanning)
```

On Debian we see 4/6:

```bash
$ nc -h  # Debian 10
OpenBSD netcat (Debian patchlevel 1.217-3)
usage: nc [-46CDdFhklNnrStUuvZz] [-I length] [-i interval] [-M ttl]
          [-m minttl] [-O length] [-P proxy_username] [-p source_port]
          [-q seconds] [-s sourceaddr] [-T keyword] [-V rtable] [-W recvlimit]
          [-w timeout] [-X proxy_protocol] [-x proxy_address[:port]]
          [destination] [port]
        Command Summary:
                -4              Use IPv4
                -6              Use IPv6
                -b              Allow broadcast
                -C              Send CRLF as line-ending
                -D              Enable the debug socket option
                -d              Detach from stdin
                -F              Pass socket fd
                -h              This help text
                -I length       TCP receive buffer length
                -i interval     Delay interval for lines sent, ports scanned
                -k              Keep inbound sockets open for multiple connects
                -l              Listen mode, for inbound connects
                -M ttl          Outgoing TTL / Hop Limit
                -m minttl       Minimum incoming TTL / Hop Limit
                -N              Shutdown the network socket after EOF on stdin
                -n              Suppress name/port resolutions
                -O length       TCP send buffer length
                -P proxyuser    Username for proxy authentication
                -p port         Specify local port for remote connects
                -q secs         quit after EOF on stdin and delay of secs
                -r              Randomize remote ports
                -S              Enable the TCP MD5 signature option
                -s sourceaddr   Local source address
                -T keyword      TOS value
                -t              Answer TELNET negotiation
                -U              Use UNIX domain socket
                -u              UDP mode
                -V rtable       Specify alternate routing table
                -v              Verbose
                -W recvlimit    Terminate after receiving a number of packets
                -w timeout      Timeout for connects and final net reads
                -X proto        Proxy protocol: "4", "5" (SOCKS) or "connect"
                -x addr[:port]  Specify proxy address and port
                -Z              DCCP mode
                -z              Zero-I/O mode [used for scanning]
        Port numbers can be individual or ranges: lo-hi [inclusive]
```

Comment (bderenzo) [reply tinystatus:32]: Default alpine linux `nc` command is from busybox.
When the `netcat-openbsd` package is installed, `nc` command supports `-4` and `-6` flags.

**Comment by bderenzo** [comment]
Fixed in 01893d422a97e62aa0ffcd4bf5f0b2614a2d3931

**Pull request closed without merging**